### PR TITLE
Feature/mcp492x

### PR DESCRIPTION
**Pull request by ctacke**

### CODE REVIEW

File: Source/Meadow.Foundation.Peripherals/ICs.ADCs.Mcp3xxx/Driver/Mcp3xxx.cs [hunk 1]
```diff
@@ -283,13 +286,15 @@ protected int ReadInternalRaw(int adcRequest, int adcResolutionInBits, int delay
                 requestBuffer[i] = (byte)(adcRequest >> (bufferSize - i - 1) * 8);
             }
 
-            spiComms.Exchange(requestBuffer, responseBuffer);
+            SpiComms.Exchange(requestBuffer, responseBuffer);
+
+            Console.WriteLine($"response: {BitConverter.ToString(responseBuffer.ToArray())}");
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Peripherals/ICs.ADCs.Mcp3xxx/Driver/Mcp3xxx.cs:291]: Remove the Console.WriteLine

File: Source/Meadow.Foundation.Peripherals/ICs.ADCs.Mcp3xxx/Driver/Mcp3xxx.AnalogInputPort.cs [hunk 2]
```diff
@@ -9,7 +9,7 @@
 
 namespace Meadow.Foundation.ICs.IOExpanders
 {
-    abstract partial class Mcp3xxx
+    public abstract partial class Mcp3xxx
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Peripherals/ICs.ADCs.Mcp3xxx/Driver/Mcp3xxx.AnalogInputPort.cs:12]: does this need to be public?

File: Source/Meadow.Foundation.Peripherals/ICs.DAC.Mcp492x/Driver/ICs.DAC.Mcp492x.csproj [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<Project Sdk="Meadow.Sdk/1.1.0">
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Peripherals/ICs.DAC.Mcp492x/Driver/ICs.DAC.Mcp492x.csproj:1]: Update metadata

File: Source/Meadow.Foundation.Peripherals/ICs.DAC.Mcp492x/Driver/Mcp4922.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
+﻿using Meadow.Hardware;
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Peripherals/ICs.DAC.Mcp492x/Driver/Mcp4922.cs:1]: Move this class into a Drivers folder (convension)

File: Source/Meadow.Foundation.Peripherals/ICs.DAC.Mcp492x/Driver/Mcp492x.cs [hunk 5]
```diff
@@ -0,0 +1,85 @@
+using Meadow.Hardware;
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Peripherals/ICs.DAC.Mcp492x/Driver/Mcp492x.cs:1]: XML comments for public things

File: Source/Meadow.Foundation.Peripherals/ICs.DAC.Mcp492x/Driver/Mcp4921.cs [hunk 6]
```diff
@@ -0,0 +1,65 @@
+﻿using Meadow.Hardware;
+using System.Collections;
+using System.Collections.Generic;
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Peripherals/ICs.DAC.Mcp492x/Driver/Mcp4921.cs:3]: XML comment all-the-things

File: Source/Meadow.Foundation.Peripherals/ICs.ADCs.Mcp3xxx/Driver/Mcp3xxx.cs (hunk 1 above)
Comment (ctacke) [reply Source/Meadow.Foundation.Peripherals/ICs.ADCs.Mcp3xxx/Driver/Mcp3xxx.cs:291]: Doh!

File: Source/Meadow.Foundation.Peripherals/ICs.ADCs.Mcp3xxx/Driver/Drivers/Mcp3201.cs [hunk 7]
```diff
@@ -73,7 +73,16 @@ public IAnalogInputPort CreateAnalogInputPort(int sampleCount, TimeSpan sampleIn
         /// <returns>A value corresponding to relative voltage level on specified device channel</returns>
         protected override int ReadInternal(int channel, InputType inputType, int adcResolutionBits)
         {
-            return ReadInternalRaw(adcRequest: 0, adcResolutionInBits: 12 + 1, delayBits: 0) >> 1;
+            Span<byte> buffer = stackalloc byte[2];
```
[the remark this answers is not in the document]
Comment (ctacke) [reply Source/Meadow.Foundation.Peripherals/ICs.ADCs.Mcp3xxx/Driver/Drivers/Mcp3201.cs:76]: Probably, but I don't have the hardware to validate that.

**Pull request merged**